keps/sig-cli/4764-kubectl-watchlist/README.md

```diff
@@ -0,0 +1,757 @@
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+### Upgrade / Downgrade Strategy
+
+This feature will not be enabled by default in kubectl until the dependent featuregate in the apiserver 
+are enabled by default.
+
+Upgrading requires that the Kubelet's environment variable be configured to `true`, and the apiserver must
+enable `WatchList` and `ConsistentListFromCache`, as well as disable `WatchFromStorageWithoutResourceVersion`. 
+
+Downgrade requires kubectl environment variable be configured to `false`.
+
+### Version Skew Strategy
+
+This feature will not be enabled by default in kubectl until the dependent featuregate in the apiserver 
+are enabled by default. However, if users actively enable it while using an older version of the apiserver,
+the following scenarios may occur:
+
+#### `WatchList` is not supported or disabled
+
+When the API is on an older version or the dependent features have not yet been enabled, the `sendInitialEvent` 
+parameter in the `WatchList` request will not be correctly processed. apiserver prior to version 1.27 will ignore 
+this parameter and treat it as a standard `watch` request, which may lead to unexpected behavior in kubectl. 
+In apiserver version 1.27 and later, if the WatchList feature is disabled, it will respond with an error as 
+expected, and then fallback to the original List logic.
+
+#### `ConsistentListFromCache` or `WatchFromStorageWithoutResourceVersion` is not supported or configured expectedly
+
+This will cause requests to directly penetrate into `etcd`, resulting in increased server-side load.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
```

Comment:
This'll need filling in at some point, but isn't required for merging as provisional.